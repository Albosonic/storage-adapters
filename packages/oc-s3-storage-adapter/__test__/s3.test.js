const s3 = require('../');

const options = {
  bucket: 'test',
  region: 'region-test',
  key: 'test-key',
  secret: 'test-secret'
};

test('validate valid conf', () => {
  const client = new s3(options);
  expect(client.isValid()).toBe(true);
});

test('validate missing bucket conf', () => {
  const missingBucketOptions = options;
  missingBucketOptions.bucket = undefined;
  const client = new s3(missingBucketOptions);
  expect(client.isValid()).toBe(false);
});
