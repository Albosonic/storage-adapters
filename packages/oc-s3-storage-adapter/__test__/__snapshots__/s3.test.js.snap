// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should expose the correct methods 1`] = `
Object {
  "adapterType": "s3",
  "getFile": [Function],
  "getJson": [Function],
  "getUrl": [Function],
  "isValid": [Function],
  "listSubDirectories": [Function],
  "maxConcurrentRequests": 20,
  "putDir": [Function],
  "putFile": [Function],
  "putFileContent": [Function],
}
`;

exports[`test getFile  1`] = `"{\\"data\\": \\"words\\"}"`;

exports[`test getJSon  1`] = `
Object {
  "data": "words",
}
`;

exports[`test getUrl  1`] = `"/test/1.0.0/test.js"`;

exports[`test listObjects  1`] = `
Array [
  "Prefix",
  "Prefix2",
]
`;

exports[`test private putFileContent  1`] = `
Object {
  "ACL": "authenticated-read",
  "Body": "words",
  "Bucket": "my-bucket",
  "ContentType": "application/javascript",
  "Expires": 2018-01-01T00:00:00.000Z,
  "Key": "filename.js",
  "ServerSideEncryption": "AES256",
}
`;

exports[`test private putFileContent  2`] = `undefined`;

exports[`test public putFileContent  1`] = `
Object {
  "ACL": "public-read",
  "Body": "words",
  "Bucket": "my-bucket",
  "ContentEncoding": "gzip",
  "Expires": 2019-01-01T00:00:00.000Z,
  "Key": "filename.gz",
  "ServerSideEncryption": "AES256",
}
`;

exports[`test public putFileContent  2`] = `undefined`;

exports[`test putFile  1`] = `
Object {
  "ACL": "public-read",
  "Bucket": "my-bucket",
  "ContentType": "application/javascript",
  "Expires": 2020-01-01T00:00:00.000Z,
  "Key": "filename.js",
  "ServerSideEncryption": "AES256",
}
`;

exports[`test putFile  2`] = `[TypeError: Cannot convert undefined or null to object]`;
